File: spaceship-bluetooth/app/src/main/java/it/scoppelletti/spaceship/bluetooth/sample/MainApp.java
package it.scoppelletti.spaceship.bluetooth.sample;

import android.app.Application;
import org.greenrobot.eventbus.EventBus;

public final class MainApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        EventBus.builder()
                .sendSubscriberExceptionEvent(false)
                .sendNoSubscriberEvent(false)
                .eventInheritance(false)
                .throwSubscriberException(BuildConfig.DEBUG)
                .addIndex(new MainEventBusIndex())
                .installDefaultEventBus();
    }
}

File: spaceship-ui/app/src/main/kotlin/it/scoppelletti/spaceship/sample/MainApp.kt
package it.scoppelletti.spaceship.sample

import android.app.Application
import com.jakewharton.threetenabp.AndroidThreeTen
import it.scoppelletti.spaceship.inject.StdlibComponent
import it.scoppelletti.spaceship.inject.UIComponent
import it.scoppelletti.spaceship.inject.UIComponentProvider
import it.scoppelletti.spaceship.sample.inject.AppComponent
import it.scoppelletti.spaceship.sample.inject.DaggerAppComponent

@Suppress("unused")
class MainApp : Application(), UIComponentProvider {

    private lateinit var _appComponent: AppComponent

    override fun onCreate() {
        super.onCreate()

        AndroidThreeTen.init(this)
        _appComponent = DaggerAppComponent.factory()
                .create(this)
    }

    override fun stdlibComponent(): StdlibComponent = _appComponent

    override fun uiComponent(): UIComponent = _appComponent
}



